native/cocos/gi/light-probe/LightProbe.cpp

```diff
@@ -209,6 +211,106 @@ void LightProbeInfo::clearSHCoefficients() {
     for (auto &probe : probes) {
         probe.coefficients.clear();
     }
+
+    clearAllSHUBOs();
+}
+
+bool LightProbeInfo::addNode(Node* node) {
+    if (!node) {
+        return false;
+    }
+
+    for (auto &item : _nodes) {
+        if (item.node == node) {
+            return false;
+        }
+    }
+
+    _nodes.emplace_back(node);
+
+    return true;
+}
+
+bool LightProbeInfo::removeNode(Node *node) {
+    if (!node) {
+        return false;
+    }
+
+    for (auto iter = _nodes.begin(); iter != _nodes.end(); ++iter) {
+        if (iter->node == node) {
+            _nodes.erase(iter);
+            return true;
+        }
+    }
+
+    return false;
+}
+
+void LightProbeInfo::syncData(Node *node, const ccstd::vector<Vec3> &probes) {
+    for (auto &item : _nodes) {
+        if (item.node == node) {
+            item.probes = probes;
+            return;
+        }
+    }
+}
+
+void LightProbeInfo::update(bool updateTet) {
+    ccstd::vector<Vec3> points;
+
+    for (auto &item : _nodes) {
+        auto *node = item.node;
+        auto &probes = item.probes;
+        const auto &worldPosition = node->getWorldPosition();
+
+        for (auto &probe : probes) {
```

Comment:
What could the size of `points` be?  Should we reserve its capacity to avoid memory re-allocation?

Reply:
Will the `update` method be invoked frequently?

Reply:
Only invoked when user click the generate probe button